torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2306,6 +2306,44 @@ def generator():
 
     return list(generator())
 
+
+def sample_inputs_conv2d(op_info, device, dtype, requires_grad, jit_fail_sample=False, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # Ordered as shapes for input, weight, bias
+    # and a dict of values of (stride, padding, groups, dilation)
+    if jit_fail_sample:
+        cases: Tuple = (
+            ((1, 4, 5, 5), (3, 4, 3, 3), None, {}),
+        )
+    else:
+        cases = (  # type: ignore[assignment]
+            ((1, 3, 4, 4), (3, 3, 3, 3), (3,),
+             {'stride': (2, 2), 'padding': 2, 'groups': 1}),
+            ((2, 4, 8, 8), (2, 2, 3, 3), (2,),
+             {'stride': (2, 2), 'padding': (2, 1), 'groups': 2, 'dilation': (4, 4)}),
+            ((1, 4, 5, 5), (1, 4, 2, 3), (1,),
+             {'stride': 2, 'padding': 1, 'groups': 1, 'dilation': (2, 3)}),
+            ((1, 4, 5, 5), (1, 4, 2, 3), (1,),
+             {'stride': 2, 'padding': 1, 'groups': 1, 'dilation': (2, 3)}),
+            ((1, 2, 4, 3), (4, 2, 3, 4), None,
+             {'stride': 2, 'padding': 1, 'groups': 1}),
+            ((1, 4, 5, 5), (1, 4, 2, 3), (1,),
+             {'stride': 2, 'padding': "valid"}),
+            ((1, 4, 5, 5), (1, 4, 2, 3), (1,),
+             {'stride': 1, 'padding': "same", 'dilation': 3}),
+        )
```

Comment:
conv2d is a pretty important op for PyTorch. Could we make sure to include all of the conv2d cases from https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_nn.py here as well?

For example, group convolution has a few different test cases: https://github.com/pytorch/pytorch/blob/3c15822f5f4ab616eb6a519a0ff9b82fc7a3dc63/torch/testing/_internal/common_nn.py#L2093-L2132 and https://github.com/pytorch/pytorch/blob/3c15822f5f4ab616eb6a519a0ff9b82fc7a3dc63/torch/testing/_internal/common_nn.py#L1997-L2015.

Reply:
Have added these cases. Took a cursory glance all other cases seem covered. Thanks!